src/meta/client/MetaClient.cpp

```diff
@@ -321,35 +321,52 @@ void MetaClient::getResponse(Request req,
                     updateActive();
                 }
                 if (retry < retryLimit) {
-                    sleep(FLAGS_meta_client_retry_interval_secs);
-                    getResponse(std::move(req), std::move(remoteFunc), std::move(respGen),
-                                std::move(p), toLeader, retry + 1, retryLimit);
+                    evb->runAfterDelay([req = std::move(req), remoteFunc = std::move(remoteFunc),
+                                        respGen = std::move(respGen), pro = std::move(pro),
+                                        toLeader, retry, retryLimit, this] () mutable {
+                        getResponse(std::move(req),
+                                    std::move(remoteFunc),
+                                    std::move(respGen),
+                                    std::move(pro),
+                                    toLeader,
+                                    retry + 1,
+                                    retryLimit);
+                    }, FLAGS_meta_client_retry_interval_secs * 1000);
+                    return;
                 } else {
-                    p.setValue(Status::Error(folly::stringPrintf("RPC failure in MetaClient: %s",
-                                                                 t.exception().what().c_str())));
+                    LOG(INFO) << "Exceed retry limit";
+                    pro.setValue(Status::Error(folly::stringPrintf("RPC failure in MetaClient: %s",
+                                                                   t.exception().what().c_str())));
                 }
                 return;
             }
             auto&& resp = t.value();
             if (resp.code == cpp2::ErrorCode::SUCCEEDED) {
                 // succeeded
-                p.setValue(respGen(std::move(resp)));
+                pro.setValue(respGen(std::move(resp)));
                 return;
             } else if (resp.code == cpp2::ErrorCode::E_LEADER_CHANGED) {
                 HostAddr leader(resp.get_leader().get_ip(), resp.get_leader().get_port());
                 {
                     folly::RWSpinLock::WriteHolder holder(hostLock_);
                     leader_ = leader;
                 }
+                if (retry < retryLimit) {
```

Comment:
Why move the retry logic code in the "else if" scope. 
All failures should be retry.

Reply:
I can hardly agree. For example, it's not appropriate to retry on `DESCRIBE TAG typo_tag_name`

Reply:
make sense

Reply:
I think the other failures should be handled in upper stream, for example, `create space existed`, it seems no necessary to retry.

Reply:
make sense